src/hocs/undo-hoc.jsx

```diff
@@ -0,0 +1,96 @@
+import bindAll from 'lodash.bindall';
+import PropTypes from 'prop-types';
+import React from 'react';
+import omit from 'lodash.omit';
+import {connect} from 'react-redux';
+
+import {getSelectedLeafItems} from '../helper/selection';
+import {setSelectedItems} from '../reducers/selected-items';
+import {performUndo, performRedo, shouldShowUndo, shouldShowRedo} from '../helper/undo';
+import {undo, redo} from '../reducers/undo';
+
+import {isBitmap} from '../lib/format';
+import Formats from '../lib/format';
+
+const UndoHOC = function (WrappedComponent) {
+    class UndoWrapper extends React.Component {
+        constructor (props) {
+            super(props);
+            bindAll(this, [
+                'handleUndo',
+                'handleRedo',
+                'handleSetSelectedItems',
+                'shouldShowUndo',
+                'shouldShowRedo'
+            ]);
+        }
+        handleUndo () {
+            performUndo(this.props.undoState, this.props.onUndo, this.handleSetSelectedItems, this.props.onUpdateImage);
+            // TODO this.handleUpdateImage
```

Comment:
Do you have a Github issue filed for this? If not, would you mind filing an issue and linking it to this `@todo` item?

Reply:
Oops, that was just a note to self that's already been fixed